internal/espressoreader/espresso_helper.go

```diff
@@ -51,7 +51,7 @@ func (eh *EspressoHelper) getL1FinalizedHeight(ctx context.Context, espressoBloc
 
 			l1FinalizedNumber := gjson.Get(espressoHeader, "fields.l1_finalized.number").Uint()
 			l1FinalizedTimestampStr := gjson.Get(espressoHeader, "fields.l1_finalized.timestamp").Str
-			if len(l1FinalizedTimestampStr) < 2 {
+			if l1FinalizedNumber == 0 || len(l1FinalizedTimestampStr) < 2 {
```

Comment:
Isn't the absence of `l1FinalizedNumber` or `l1FinalizedTimestampStr` an error? I mean, it will never be fixed, since Espresso's blocks are definitive. I'd even say the reader is kind of doomed in this situation, because it will never be able to recover. Just logging a debug info is too weak, is must be clear that it's stuck forever.

Reply:
I see. Still, if the header is there but it's missing some info or has some invalid stuff, it would be an error (or at least a warning) wouldn't it? I just want to avoid the situation in which the thing is frozen (using the dev node) and there is no indication of it